tiledb/api/src/array/enumeration/strategy.rs

```diff
@@ -0,0 +1,161 @@
+use std::cmp::Ordering;
+
+use proptest::collection::vec;
+use proptest::prelude::*;
+
+use crate::array::EnumerationData;
+use crate::datatype::strategy::*;
+use crate::{fn_typed, Datatype};
+
+pub fn prop_enumeration_name() -> impl Strategy<Value = String> {
+    proptest::string::string_regex("[a-zA-Z0-9_]+")
+        .expect("Error creating attribute name strategy")
+}
+
+fn prop_cell_val_num() -> impl Strategy<Value = Option<u32>> {
+    Just(None)
+}
+
+fn prop_ordered() -> impl Strategy<Value = bool> {
+    any::<bool>()
+}
+
+trait WtfNumCmp: std::marker::Sized {
+    fn compare(&self, other: &Self) -> Ordering;
+}
+
+macro_rules! derive_wtf_cmp {
+    ($typename:ident) => {
+        impl WtfNumCmp for $typename {
+            fn compare(&self, other: &Self) -> Ordering {
+                std::cmp::Ord::cmp(self, &other)
+            }
+        }
+    };
+}
+
+derive_wtf_cmp!(u8);
+derive_wtf_cmp!(u16);
+derive_wtf_cmp!(u32);
+derive_wtf_cmp!(u64);
+derive_wtf_cmp!(i8);
+derive_wtf_cmp!(i16);
+derive_wtf_cmp!(i32);
+derive_wtf_cmp!(i64);
+
+impl WtfNumCmp for f32 {
+    fn compare(&self, other: &Self) -> Ordering {
+        // Handle either is a NaN
```

Comment:
For this particular case that shouldn't matter since the sort is followed immediately followed by dedup which will only leave a single NaN. And then beyond that, Enumeration values are internally treated as opaque bytes without interpreting so even having multiple NaN values with different bit level representations won't change the behavior.

Reply:
Fixed with the AnyNumCmp trait.